cdc/processor/sourcemanager/engine/pebble/event_sorter.go

```diff
@@ -280,6 +335,9 @@ func (s *EventIter) Next() (event *model.PolymorphicEvent, pos engine.Position,
 		}
 		event, s.headItem = s.headItem, event
 	}
+	if event != nil {
+		s.state.remainEvents.Add(-1)
```

Comment:
If redo is enabled, an event can be pushed 1 time but fetched 2 times.
So I suggest to record input events in `SortEngine`, and record fetched events in `tableSinkWrapper`.